In [1]:
print("hello")

hello


In [2]:
import fitz  # PyMuPDF

pdf_path = "../data/raw/sample_medical_report.pdf"

In [3]:
pdf_document = fitz.open(pdf_path)
print("pages:", len(pdf_document))

pages: 1


In [4]:
for page_num in range(len(pdf_document)):
    print(page_num)

0


In [5]:
page  = pdf_document.load_page(0)
text = page.get_text()
print(text)

MEDICAL LABORATORY REPORT
Patient Information
Patient Name: John Smith
Age: 45 Years
Gender: Male
Patient ID: P123456
Date of Test: 05 June 2026
Complete Blood Count (CBC)
Hemoglobin (Hb): 11.2 g/dL (Normal: 13.5 - 17.5 g/dL)
White Blood Cells (WBC): 12,500 /µL (Normal: 4,000 - 11,000 /µL)
Platelets: 250,000 /µL (Normal: 150,000 - 450,000 /µL)
Blood Sugar
Fasting Blood Sugar: 145 mg/dL (Normal: 70 - 99 mg/dL)
HbA1c: 7.2% (Normal: Below 5.7%)
Lipid Profile
Total Cholesterol: 230 mg/dL
LDL Cholesterol: 150 mg/dL
HDL Cholesterol: 38 mg/dL
Triglycerides: 220 mg/dL
Kidney Function Test
Creatinine: 1.3 mg/dL
Blood Urea Nitrogen (BUN): 24 mg/dL
Doctor's Notes
Patient shows elevated blood glucose levels suggesting Type 2 Diabetes. Cholesterol levels are
above the recommended range. Mild anemia is present due to low hemoglobin levels. Follow-up
consultation recommended.



In [6]:
len(text)

874

In [ ]:
full_text = ""
for page_num in range(len(pdf_document)):
    page  = pdf_document.load_page(page_num)
    text = page.get_text()
    full_text += text + "\n"

    pdf_document.close()

In [15]:
print(full_text[:800])  # Print the first 1000 characters of the full text

MEDICAL LABORATORY REPORT
Patient Information
Patient Name: John Smith
Age: 45 Years
Gender: Male
Patient ID: P123456
Date of Test: 05 June 2026
Complete Blood Count (CBC)
Hemoglobin (Hb): 11.2 g/dL (Normal: 13.5 - 17.5 g/dL)
White Blood Cells (WBC): 12,500 /µL (Normal: 4,000 - 11,000 /µL)
Platelets: 250,000 /µL (Normal: 150,000 - 450,000 /µL)
Blood Sugar
Fasting Blood Sugar: 145 mg/dL (Normal: 70 - 99 mg/dL)
HbA1c: 7.2% (Normal: Below 5.7%)
Lipid Profile
Total Cholesterol: 230 mg/dL
LDL Cholesterol: 150 mg/dL
HDL Cholesterol: 38 mg/dL
Triglycerides: 220 mg/dL
Kidney Function Test
Creatinine: 1.3 mg/dL
Blood Urea Nitrogen (BUN): 24 mg/dL
Doctor's Notes
Patient shows elevated blood glucose levels suggesting Type 2 Diabetes. Cholesterol levels are
above the recommended range. Mild anemia is 


In [16]:
len(full_text)

875

In [17]:
words  = full_text.split()
len(words)

132

In [19]:
with open(
    "../artifacts/extracted_text/report.txt", "w",
    encoding="utf-8"
) as f:
    f.write(full_text)

In [21]:
with open(
    "../artifacts/extracted_text/report.txt", "r",
    encoding="utf-8"
) as f:
    extracted_text = f.read()

print(extracted_text[:800])  # Print the first 1000 characters of the extracted text

MEDICAL LABORATORY REPORT
Patient Information
Patient Name: John Smith
Age: 45 Years
Gender: Male
Patient ID: P123456
Date of Test: 05 June 2026
Complete Blood Count (CBC)
Hemoglobin (Hb): 11.2 g/dL (Normal: 13.5 - 17.5 g/dL)
White Blood Cells (WBC): 12,500 /µL (Normal: 4,000 - 11,000 /µL)
Platelets: 250,000 /µL (Normal: 150,000 - 450,000 /µL)
Blood Sugar
Fasting Blood Sugar: 145 mg/dL (Normal: 70 - 99 mg/dL)
HbA1c: 7.2% (Normal: Below 5.7%)
Lipid Profile
Total Cholesterol: 230 mg/dL
LDL Cholesterol: 150 mg/dL
HDL Cholesterol: 38 mg/dL
Triglycerides: 220 mg/dL
Kidney Function Test
Creatinine: 1.3 mg/dL
Blood Urea Nitrogen (BUN): 24 mg/dL
Doctor's Notes
Patient shows elevated blood glucose levels suggesting Type 2 Diabetes. Cholesterol levels are
above the recommended range. Mild anemia is 


Observations:
1. PDF extraction successful
2. Total pages : 1
3. Total words : 132
4. Any formating is lost. Only text is extracted.
5. tables executed correctly ? yes